src/Microsoft.QualityGuidelines.Analyzers/Core/MarkMembersAsStatic.Fixer.cs

```diff
@@ -22,9 +29,33 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await Task.Run(() => { }).ConfigureAwait(false);
-            throw new NotImplementedException();
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span);
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var symbol = model.GetDeclaredSymbol(node, context.CancellationToken);
+
+            context.RegisterCodeFix(new MarkMembersAsStaticAction(
+                    string.Format(MicrosoftQualityGuidelinesAnalyzersResources.MarkMembersAsStaticCodeFix, symbol.Name),
+                    async ct => await RemoveStaticModifier(context.Document, root, node, ct).ConfigureAwait(false)
+                ), context.Diagnostics.Single());
+        }
+
+        private Task<Document> RemoveStaticModifier(Document document, SyntaxNode root, SyntaxNode node, CancellationToken ct)
+        {
+            ct.ThrowIfCancellationRequested();
+            var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+            var madeStatic = syntaxGenerator.WithModifiers(node, DeclarationModifiers.Static);
+            SyntaxNode newRoot = root.ReplaceNode(node, madeStatic);
+            return Task.FromResult(document.WithSyntaxRoot(newRoot));
+        }
+
+        private class MarkMembersAsStaticAction : DocumentChangeAction
+        {
+            public MarkMembersAsStaticAction(string title, Func<CancellationToken, Task<Document>> createChangedDocument) : base(title, createChangedDocument)
+            {
+            }
+
+            public override string EquivalenceKey => Title;
```

Comment:
this seems a bit dangerous. unless we do 1) or 2), I think we shouldn't support batch fixing.